cmd/copy.go

```diff
@@ -267,6 +274,81 @@ func (raw rawCopyCmdArgs) stripTrailingWildcardOnRemoteSource(location common.Lo
 	return
 }
 
+// performNFSSpecificValidation performs validation specific to NFS (Network File System) configurations
+// for a synchronization command. It checks NFS-related flags and settings and ensures that the necessary
+// properties are set correctly for NFS copy operations.
+//
+// The function checks the following:
+//   - Validates the "preserve-nfs-info" flag to ensure it is set correctly for NFS-aware locations.
+//   - Validates the "preserve-nfs-permissions" flag, ensuring that user input is correct and provides feedback
+//     if the flag is set to false and NFS info is being preserved.
+//   - Ensures that both source and destination locations are NFS-aware for relevant operations.
+//
+// If any of these validations fail, the function returns an error with the respective validation message.
+//
+// Returns:
+// - An error if any validation fails, otherwise nil indicating successful validation.
+func (raw rawCopyCmdArgs) performNFSSpecificValidation(cooked *CookedCopyCmdArgs) (err error) {
+	// TODO remove it
+	glcm.Info("NFS copy support addition in progress")
+	cooked.isNFSCopy = raw.isNFSCopy
+	cooked.preserveNFSInfo = raw.preserveNFSInfo && areBothLocationsNFSAware(cooked.FromTo)
```

Comment:
sidenote: I like this pattern for the "cook" step-- we should continue doing this in the future; separating individual steps into their own functions.